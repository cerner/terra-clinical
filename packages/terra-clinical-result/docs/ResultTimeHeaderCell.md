# Result Time Header Cell

Result-Time-Header-Cell is a component of terra-clinical-result that creates a standardized view for the Date/Time Header cells (column headers) for each bucketed time column within a flowsheet.

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-clinical-result`


<!-- AUTO-GENERATED-CONTENT:START Peer Dependencies -->
<!-- AUTO-GENERATED-CONTENT:END -->


## Component Features
