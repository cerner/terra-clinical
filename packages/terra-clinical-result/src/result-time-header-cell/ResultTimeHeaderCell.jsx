import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './ResultTimeHeaderCell.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Content to be displayed on the first line, typically abbrivated date, e.g. `Dec 12, 2010`
   */
  date: PropTypes.string.isRequired,
  /**
   * Content to be displayed on the second line, typically 24 hour time in hours and minutes, e.g. `23:59`
   */
  time: PropTypes.string.isRequired,
  /**
   * Visually hides the date when presented in a series of side-by-side columns of the same date.
   */
  hideDate: PropTypes.bool,
  /**
   * The padding styling to apply to the Result Time column header cell.
   * One of `'none'`, `'compact'`, or `'standard'`.
  */
  paddingStyle: PropTypes.oneOf(['none', 'compact', 'standard']),
};

const defaultProps = {
  paddingStyle: 'compact',
};

const ResultTimeHeaderCell = (props) => {
  const {
    date,
    time,
    hideDate,
    paddingStyle,
    ...customProps
  } = props;

  const dateClassnames = cx([
    'date',
    { 'hide-date': hideDate },
  ]);

  const timeHeaderCellClassnames = cx([
    'clinical-result-time-header-cell',
    { 'padding-standard': paddingStyle === 'standard' },
    { 'padding-compact': paddingStyle === 'compact' },
  ]);

  return (
    <div
      {...customProps}
      className={customProps.className ? `${timeHeaderCellClassnames} ${customProps.className}` : timeHeaderCellClassnames}
    >
      <div className={dateClassnames}>{date}</div>
      <div className={cx('time')}>{time}</div>
    </div>
  );
};

ResultTimeHeaderCell.propTypes = propTypes;
ResultTimeHeaderCell.defaultProps = defaultProps;

export default ResultTimeHeaderCell;
