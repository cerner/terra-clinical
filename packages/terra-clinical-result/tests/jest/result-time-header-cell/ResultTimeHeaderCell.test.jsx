import React from 'react';
import ResultTimeHeaderCell from '../../../src/result-time-header-cell/ResultTimeHeaderCell';

// Snapshot Tests
describe('ResultTimeHeaderCell', () => {
  it('should render a date and time', () => {
    const cell = shallow(<ResultTimeHeaderCell date="December 31st 1999" time="10:00 PM" />);
    expect(cell).toMatchSnapshot();
  });

  it('should hide date', () => {
    const cell = shallow(<ResultTimeHeaderCell date="December 31st 1999" time="10:00 PM" hideDate />);
    expect(cell).toMatchSnapshot();
  });

  describe('paddingStyle -', () => {
    it('should render with none', () => {
      const cell = shallow(<ResultTimeHeaderCell date="December 31st 1999" time="10:00 PM" paddingStyle="none" />);
      expect(cell).toMatchSnapshot();
    });

    it('should render with standard', () => {
      const cell = shallow(<ResultTimeHeaderCell date="December 31st 1999" time="10:00 PM" paddingStyle="standard" />);
      expect(cell).toMatchSnapshot();
    });

    it('should render with compact', () => {
      const cell = shallow(<ResultTimeHeaderCell date="December 31st 1999" time="10:00 PM" paddingStyle="compact" />);
      expect(cell).toMatchSnapshot();
    });
  });
});
