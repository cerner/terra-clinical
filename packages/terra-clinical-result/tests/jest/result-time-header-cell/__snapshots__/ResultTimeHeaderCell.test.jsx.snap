// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultTimeHeaderCell paddingStyle - should render with compact 1`] = `
<div
  className="clinical-result-time-header-cell padding-compact"
>
  <div
    className="date"
  >
    December 31st 1999
  </div>
  <div
    className="time"
  >
    10:00 PM
  </div>
</div>
`;

exports[`ResultTimeHeaderCell paddingStyle - should render with none 1`] = `
<div
  className="clinical-result-time-header-cell"
>
  <div
    className="date"
  >
    December 31st 1999
  </div>
  <div
    className="time"
  >
    10:00 PM
  </div>
</div>
`;

exports[`ResultTimeHeaderCell paddingStyle - should render with standard 1`] = `
<div
  className="clinical-result-time-header-cell padding-standard"
>
  <div
    className="date"
  >
    December 31st 1999
  </div>
  <div
    className="time"
  >
    10:00 PM
  </div>
</div>
`;

exports[`ResultTimeHeaderCell should hide date 1`] = `
<div
  className="clinical-result-time-header-cell padding-compact"
>
  <div
    className="date hide-date"
  >
    December 31st 1999
  </div>
  <div
    className="time"
  >
    10:00 PM
  </div>
</div>
`;

exports[`ResultTimeHeaderCell should render a date and time 1`] = `
<div
  className="clinical-result-time-header-cell padding-compact"
>
  <div
    className="date"
  >
    December 31st 1999
  </div>
  <div
    className="time"
  >
    10:00 PM
  </div>
</div>
`;
